### PR TITLE
Document facet types

### PR DESCRIPTION
**Pull request by MatMoore**
Add documentation to the `type` section and add sections for `open_value` and `closed_value`.

This functionality was added for the topical events finder: https://www.gov.uk/government/topical-events

Rendered markdown:
https://github.com/alphagov/finder-frontend/blob/4716d57bc69d48a3ea77ba7db9ff5d7881cdbe7c/docs/finder-content-item.md

### CODE REVIEW

**Comment by tijmenb** [comment]
The structure of the docs make it hard to follow (was like this before). Maybe we could add an examples to the `facets` chapter?

https://github.com/alphagov/finder-frontend/blob/4716d57bc69d48a3ea77ba7db9ff5d7881cdbe7c/docs/finder-content-item.md#facets

**Comment by MatMoore** [comment]
@tijmenb I've rewritten this. What do you think? https://github.com/alphagov/finder-frontend/blob/65ab57a34d9599dba18d42095147b2c11e9cbe89/docs/finder-content-item.md#facets

Will swash the commits before merging.

File: docs/finder-content-item.md [hunk 1]
```diff
@@ -99,72 +99,195 @@ A string. Optional.
...
+  "key": "case_type",
+  "name": "Case type",
+  "type": "text",
+  "display_as_result_metadata": true,
+  "filterable": true,
+  "allowed_values": [...],
+  [...]
+}
+```
+<img alt="Filterable facets" src="./assets/filter.png" width="260" height="273">
```
Comment (tijmenb) [inline docs/finder-content-item.md]: You can use `Label` here

**Comment by tijmenb** [comment]
Awesome!

File: docs/finder-content-item.md (hunk 1 above)
Comment (Davidslv) [inline docs/finder-content-item.md]: They changed the functionality, now you forced to use HTML tags instead if you need resizing http://stackoverflow.com/a/30151727

**Comment by MatMoore** [comment]
Have squashed all the commits.
